itest/lnd_payment_test.go

```diff
@@ -767,3 +770,128 @@ func assertChannelState(ht *lntest.HarnessTest, hn *node.HarnessNode,
 	}, lntest.DefaultTimeout)
 	require.NoError(ht, err, "timeout while chekcing for balance")
 }
+
+// testPaymentFailureReasonCanceled ensures that the cancellation of a
+// SendPayment request results in the payment failure reason
+// FAILURE_REASON_CANCELED. This failure reason indicates that the context was
+// cancelled manually by the user. It does not interrupt the current payment
+// attempt, but will prevent any further payment attempts. The test steps are:
+// 1.) Alice pays Carol's invoice through Bob.
+// 2.) Bob intercepts the htlc, keeping the payment pending.
+// 3.) Alice cancels the payment context, the payment is still pending.
+// 4.) Bob fails the payment.
+// 5.) Alice observes a failed payment with failure reason
+// FAILURE_REASON_CANCELED which suppresses further payment attempts.
+func testPaymentFailureReasonCanceled(ht *lntest.HarnessTest) {
```

Comment:
it would be great to have two sub-tests, one where we test a payment failure and one that tests a success after cancellation, to document that this is known behavior

Reply:
That's a good idea. I thought of leaving a comment that the behavior in the resume and fail case are the same, but better to explicitly test for that.

Reply:
Done